### PR TITLE
fix for older linuxes where gpasswd doesn't have the relevant options…

### PR DESCRIPTION
**Pull request by dheule**
fixes Issue #395

### CODE REVIEW

**Comment by telmich** [comment]
I wonder - should we user usermod in any case instead of gpasswd? @asteven what do you think?

I think it is an appropriate patch, but I try to minimise the number of cases for special OS, if possible.

**Comment by dheule** [comment]
usermod has also different incompatible parameter Syntax from old to new
linuxes ;-((
Am 08.05.2015 15:54 schrieb "Nico Schottelius" <notifications@github.com>:

**Comment by dheule** [comment]
I had an idea, pleas wait for the next update off the pullrequest. Tank
you. Daniel
Am 08.05.2015 15:54 schrieb "Nico Schottelius" <notifications@github.com>:

**Comment by dheule** [comment]
Now i have implemented an explorer for old usermod implementations.
the switch from usermod and gpasswd is relevant because new usermod commands 
have the ability to append some groups to a user, but i don't have found an option 
to remove only one group from a user, so in the old world we must use usermod, in the newer ones we must use gpasswd.
But now we have no more a hardcoded os'es ...

Please review ...

File: cdist/conf/type/__user_groups/gencode-remote [hunk 1]
```diff
@@ -20,6 +20,17 @@
...
 user="$(cat "$__object/parameter/user" 2>/dev/null || echo "$__object_id")"
 state_should="$(cat "$__object/parameter/state")"
+oldusermod="$(cat "$__object/explorer/oldusermod")"
+
+if [ "$oldusermod" = "true" ]; then
+        old=yes
+        addparam="-A"
+        delparam="-R"
+else
+        old=no
```
Comment (telmich) [inline cdist/conf/type/__user_groups/gencode-remote]: Minor improvement possible: if we have $oldusermod, we do not need to setup old=, but can use $oldusermod below again :-)

Comment (dheule) [inline cdist/conf/type/__user_groups/gencode-remote]: Done ...

2015-05-11 10:14 GMT+02:00 Nico Schottelius <notifications@github.com>:

Comment (dheule) [inline cdist/conf/type/__user_groups/gencode-remote]: Ping ?

2015-05-11 10:19 GMT+02:00 Daniel Heule <daniel.heule@gmail.com>:

**Comment by dheule** [comment]
hello nico,
is something against a merge of this fix in the current version ?

please comment ;-)

**Comment by telmich** [comment]
Everything fine :-)

**Pull request merged**